dlt/destinations/impl/destination/factory.py

```diff
@@ -0,0 +1,114 @@
+import typing as t
+import inspect
+from importlib import import_module
+
+from types import ModuleType
+from dlt.common.typing import AnyFun
+
+from dlt.common.destination import Destination, DestinationCapabilitiesContext
+from dlt.common.configuration import known_sections, with_config, get_fun_spec
+from dlt.common.configuration.exceptions import ConfigurationValueError
+
+from dlt.destinations.impl.destination.configuration import (
+    SinkClientConfiguration,
+    TSinkCallable,
+)
+from dlt.destinations.impl.destination import capabilities
+from dlt.common.data_writers import TLoaderFileFormat
+from dlt.common.utils import get_callable_name
+
+if t.TYPE_CHECKING:
+    from dlt.destinations.impl.destination.sink import SinkClient
+
+
+class DestinationInfo(t.NamedTuple):
+    """Runtime information on a discovered destination"""
+
+    SPEC: t.Type[SinkClientConfiguration]
+    f: AnyFun
+    module: ModuleType
+
+
+_DESTINATIONS: t.Dict[str, DestinationInfo] = {}
+"""A registry of all the decorated destinations"""
+
+
+class destination(Destination[SinkClientConfiguration, "SinkClient"]):
+    def capabilities(self) -> DestinationCapabilitiesContext:
+        return capabilities(
+            self.config_params.get("loader_file_format", "puae-jsonl"),
+            self.config_params.get("naming_convention", "direct"),
+        )
+
+    @property
+    def spec(self) -> t.Type[SinkClientConfiguration]:
+        """A spec of destination configuration resolved from the sink function signature"""
+        return self._spec
+
+    @property
+    def client_class(self) -> t.Type["SinkClient"]:
+        from dlt.destinations.impl.destination.sink import SinkClient
+
+        return SinkClient
+
+    def __init__(
+        self,
+        destination_callable: t.Union[TSinkCallable, str] = None,  # noqa: A003
```

Comment:
we can still configure this right?

Reply:
as in give a function name / module path in a config bar? then yes, see the tests.